### PR TITLE
notepad4: Add version 24.07r5332

### LINKED ISSUE #13655: [Request]: notepad4
### Prerequisites

- [X] I have searched all issues/PRs to ensure it has not already been reported or fixed.

### Criteria

- [X] Reasonably well-known and widely used (e.g. if it's a GitHub project, it should have at least 100 stars and/or 50 forks)
- [X] English interface (or at least English documentation)
- [X] Latest stable version
- [X] Full version (i.e. not a trial version)
- [X] Fairly standard install (e.g. uses a version-specific download URL, no elaborate pre/post install scripts)

### Name

notepad4

### Description

Fork of Notepad2, a light-weight Scintilla-based text editor. Featuring syntax highlighting, code folding, auto-completion and API list for about 80 programming languages/documents. Successor of notepad-zufuliu.

### Homepage

https://github.com/zufuliu/notepad4

### Download Link(s)

https://github.com/zufuliu/notepad4/releases/download/v24.07r5332/Notepad4_HD_i18n_AVX2_v24.07r5332.zip

### Some Indication of Popularity/Repute

Successor of notepad-zufuliu

### PR DESCRIPTION
**Pull request by chemPolonium**
Closes #13655

Relates to #13655

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

File: bucket/notepad4.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        "64bit": {
+            "url": "https://github.com/zufuliu/notepad4/releases/download/v24.07r5332/Notepad4_HD_i18n_AVX2_v24.07r5332.zip",
+            "hash": "698ba0e3dfe032ed0ae65dcb2559c6a1150db1d0bf3105f133c0af98158266a7"
+        },
+        "32bit": {
+            "url": "https://github.com/zufuliu/notepad4/releases/download/v24.07r5332/Notepad4_i18n_Win32_v24.07r5332.zip",
+            "hash": "6ad13bd7114bbc39c708eeee8de2d9717abf0fedf6a98165a2e13449d33be0dd"
+        }
+    },
+    "bin": "notepad2.exe",
```
Comment (aliesbelik) [inline bucket/notepad4.json:16]:
```suggestion
    "bin": "Notepad4.exe",
```

File: bucket/notepad4.json [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    "shortcuts": [
+        [
+            "Notepad4.exe",
+            "Notepad4"
+        ],
+        [
+            "metapath.exe",
+            "metapath"
```
Comment (aliesbelik) [inline bucket/notepad4.json:24]:
```suggestion
            "matepath.exe",
            "matepath"
```

File: bucket/notepad4.json [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        ]
+    ],
+    "persist": [
+        "Notepad4.ini",
+        "metapath.ini"
```
Comment (aliesbelik) [inline bucket/notepad4.json:29]:
```suggestion
        "matepath.ini"
```

**Review by aliesbelik — CHANGES_REQUESTED** [review]
Please double check binaries names.

**Comment by aliesbelik** [comment]
Please add arm64 version as well.

**Comment by aliesbelik** [comment]
Manifest for [notepad2-zufuliu](https://github.com/ScoopInstaller/Extras/blob/master/bucket/notepad2-zufuliu.json) already exists in Extras and points to the same (renamed) project.
Shouldn't it be updated and renamed instead of creating new one?

**Comment by chemPolonium** [comment]
Hello, thank you for making the changes. I apologize for the incorrect file name. The project has been renamed, and the executable file name has also been changed from notepad2.exe to notepad4.exe. Therefore, updating directly on the original file might cause some compatibility issues. Can I refer to the situation with [latex](https://github.com/ScoopInstaller/Main/blob/master/bucket/latex.json) and [miktex](https://github.com/ScoopInstaller/Main/blob/master/bucket/miktex.json), modifying the instructions for notepad2-zufuliu and adding a new section for notepad4?

**Pull request closed without merging**